include/net/net_ip.h

```diff
@@ -488,7 +488,7 @@ static inline bool net_is_my_ipv4_addr(const struct in_addr *addr)
 static inline bool net_ipv4_addr_cmp(const struct in_addr *addr1,
 				     const struct in_addr *addr2)
 {
-	return addr1->s_addr == addr2->s_addr;
+	return !memcmp(addr1, addr2, sizeof(struct in_addr));
```

Comment:
Why not use UNALIGNED_GET() here? I see that you made it similar as IPv6 cmp, but for IPv4 we also have an option to use this macro instead.